pootle/static/js/editor/app.js

```diff
@@ -385,40 +403,38 @@ PTL.editor = {
     /* History support */
     $.history.init((hash) => {
       const params = utils.getParsedHash(hash);
-      let isInitial = true;
-      let uId = 0;
-      let initialOffset = 0;
 
       if (this.selectedSuggestionId !== undefined) {
         this.closeSuggestion({ checkIfCanNavigate: false });
       }
 
       // Walk through known filtering criteria and apply them to the editor object
 
+      let uid;
       if (params.unit) {
-        const uIdParam = parseInt(params.unit, 10);
-
-        if (uIdParam && !isNaN(uIdParam)) {
-          const current = this.units.getCurrent();
-          const newUnit = this.units.get(uIdParam);
+        uid = parseInt(params.unit, 10);
 
-          if (newUnit && newUnit !== current) {
-            this.setUnit(newUnit);
-            return;
+        if (uid && !isNaN(uid)) {
+          if (uid !== this.units.uid) {
+            setTimeout(() => {
```

Comment:
I might be failing to see some detail, but I don't see this being necessary. FWIW I haven't experienced any issues by taking `setTimeout` out. Related to this block, the nested `if` can be merged with the `if` at the upper level.